spec/workers/notifications/new_reaction_worker_spec.rb

```diff
@@ -0,0 +1,31 @@
+require "rails_helper"
+
+RSpec.describe Notifications::NewReactionWorker, type: :worker do
+  let(:reaction_data) { { reactable_type: "Comment", reactable_id: 1, reactable_user_id: 2 } }
+  let(:org) { create(:organization) }
+  let(:receiver_data) { { klass: "Organization", id: org.id } }
+  let(:worker) { subject }
+
+  include_examples "#enqueues_on_correct_queue", "medium_priority", [{}, {}, true]
+
+  describe "#perform" do
+    let(:reaction_service) { double }
+
+    before { allow(reaction_service).to receive(:call) }
+
+    it "calls the service" do
+      subject.perform(reaction_data, receiver_data, reaction_service)
+      expect(reaction_service).to have_received(:call).with(reaction_data, org).once
+    end
+
+    it "doesn't call if is a receiver is of a wrong class" do
+      subject.perform(reaction_data, { klass: "Tag", id: 10 }, reaction_service)
+      expect(reaction_service).not_to have_received(:call)
+    end
+
+    it "doesn't call if is a receiver doesn't exist" do
+      subject.perform(reaction_data, { klass: "Organization", id: org.id + 1 }, reaction_service)
```

Comment:
keep in mind that id + 1 could technically exist, you can pass nil to make sure it doesn't exist